src/devices/Board/Board.cs

```diff
@@ -0,0 +1,708 @@
+        private readonly object _pinReservationsLock;
+        private readonly Dictionary<int, List<PinReservation>> _pinReservations;
+        private readonly Dictionary<int, I2cBusManager> _i2cBuses;
+        private bool _initialized;
+        private bool _disposed;
+
+        /// <summary>
+        /// Constructs a board instance with the given default numbering scheme.
+        /// All methods will use the given numbering scheme, unless another scheme is explicitly given in a call.
+        /// </summary>
+        /// <param name="defaultNumberingScheme">Default numbering scheme for the board.</param>
+        /// <remarks>
+        /// The constructor will never throw an exception. Call <see cref="Initialize"/> to initialize the hardware and check
+        /// whether an instance can actually run on the current hardware.
+        /// </remarks>
+        protected Board(PinNumberingScheme defaultNumberingScheme)
+        {
+            _defaultNumberingScheme = defaultNumberingScheme;
+            _pinReservations = new Dictionary<int, List<PinReservation>>();
+            _i2cBuses = new Dictionary<int, I2cBusManager>();
+            _pinReservationsLock = new object();
+            _initialized = false;
+            _disposed = false;
+        }
+
+        /// <summary>
+        /// (Temporary) Register to this event to receive log messages.
+        /// </summary>
+        public event Action<string, Exception?>? LogMessages;
+
+        /// <summary>
+        /// True if the board instance is initialized
+        /// </summary>
+        protected bool Initialized => _initialized;
+
+        /// <summary>
+        /// True if this instance is disposed.
+        /// Any attempt to use it after this becomes true results in undefined behavior.
+        /// </summary>
+        protected bool Disposed
+        {
+            get
+            {
+                return _disposed;
+            }
+        }
+
+        /// <summary>
+        /// The default pin numbering scheme
+        /// </summary>
+        public PinNumberingScheme DefaultPinNumberingScheme
+        {
+            get
+            {
+                return _defaultNumberingScheme;
+            }
+        }
+
+        /// <summary>
+        /// (Temporary) Log a message
+        /// </summary>
+        protected void Log(string message, Exception? exception = null)
+        {
+            LogMessages?.Invoke(message, null);
+        }
+
+        /// <summary>
+        /// Converts pin numbers from one numbering scheme to another.
+        /// </summary>
+        /// <param name="pinNumber">Pin number to convert</param>
+        /// <param name="inputScheme">The numbering scheme of the input pin</param>
+        /// <param name="outputScheme">The desired numbering scheme</param>
+        /// <returns>The converted pin number</returns>
+        public abstract int ConvertPinNumber(int pinNumber, PinNumberingScheme inputScheme, PinNumberingScheme outputScheme);
```

Comment:
you can go with two designs here:
- take PinNumberingScheme as input and assume inputScheme is whatever was passed in everywhere
- PinNumberingScheme becomes part of every API taking pinNumber

currently you mixed both. My preference is first option but we can discuss it during the triage meeting